### PR TITLE
Fix misc TypeScript errors 11

### PR DESCRIPTION
**Pull request by sirbrillig**
#### Changes proposed in this Pull Request

This fixes misc. TypeScript errors in `client`.

#### Testing instructions

Verify that none of these changes will actually affect behavior of the code.

### CODE REVIEW

File: client/components/jetpack/server-credentials-wizard-dialog/index.tsx [hunk 1]
```diff
@@ -96,7 +97,6 @@ const ServerCredentialsWizardDialog = ( {
 						{ translate( 'Need help? Find your server credentials' ) }
 					</ExternalLink>
 					<ServerCredentialsForm
-						className="server-credentials-wizard-dialog__form"
```
Comment (noahtallen) [inline client/components/jetpack/server-credentials-wizard-dialog/index.tsx:99]: Does this classname need to be added back somewhere else?

**Review by noahtallen — APPROVED** [review]
Looks pretty good to me! Thanks for doing all of these :)

File: client/components/jetpack/server-credentials-wizard-dialog/index.tsx (hunk 1 above)
Comment (sirbrillig) [reply client/components/jetpack/server-credentials-wizard-dialog/index.tsx:99]: I don't know, but it's not doing anything currently so I figure that it's not required.

Comment (sirbrillig) [reply client/components/jetpack/server-credentials-wizard-dialog/index.tsx:99]: (It's also not styled anywhere that I can find.)

**Pull request merged**